engine-tests/src/test/java/org/terasology/config/flexible/validators/RangedNumberValidatorTest.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.config.flexible.validators;
+
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Enclosed.class)
+public class RangedNumberValidatorTest {
+    public static class IntegerValidator {
+        private RangedNumberValidator<Integer> validator;
+
+        private void initValidator(Integer min, Integer max, boolean minInclusive, boolean maxInclusive) {
+            validator = new RangedNumberValidator<>(min, max, minInclusive, maxInclusive);
+        }
+
+        @Test
+        public void testAllInclusive() {
+            initValidator(0, 100, true, true);
+
+            for (int i = validator.getMin(); i <= validator.getMax(); i++) {
+                assertTrue(String.format("%d returned invalid", i), validator.fastValidate(i));
+            }
```

Comment:
Looping through all 100 numbers seems a bit overcomplicated here and below. Just check a much smaller range, 1 to 3 for example, using simple, straightforward code.